alpha/apps/kaltura/lib/storage/kFileSyncUtils.class.php

```diff
@@ -2229,14 +2229,34 @@ public static function doAllEntryFlavorsExistInStorage($storageId, $flavorAssets
 	 */
 	public static function getReadyInternalFileSyncsForKey(FileSyncKey $syncKey)
 	{
-		$peridoicStorageFileSyncs = self::getFileSyncsFromPeriodicStorage($syncKey->getPartnerId(), $syncKey);
+		$periodicStorageFileSyncs = self::getFileSyncsFromPeriodicStorage($syncKey->getPartnerId(), $syncKey);
 
 		$c = FileSyncPeer::getCriteriaForFileSyncKey( $syncKey );
 		$c->addAnd ( FileSyncPeer::FILE_TYPE , FileSync::FILE_SYNC_FILE_TYPE_URL, Criteria::NOT_EQUAL);
 		$c->addAnd ( FileSyncPeer::STATUS , FileSync::FILE_SYNC_STATUS_READY );
 		$localfileSyncs = FileSyncPeer::doSelect( $c );
 
-		return array_merge($peridoicStorageFileSyncs, $localfileSyncs);
+		return array_merge($periodicStorageFileSyncs, $localfileSyncs);
+	}
+	
+	public static function getReadyRemoteFileSyncsForAsset($id, $asset, $objectType, $objectSubType)
+	{
+		$c = new Criteria();
+		$c->add(FileSyncPeer::OBJECT_TYPE, $objectType);
+		$c->add(FileSyncPeer::OBJECT_SUB_TYPE, $objectSubType);
+		$c->add(FileSyncPeer::OBJECT_ID, $id);
+		$c->add(FileSyncPeer::VERSION, $asset->getVersion());
+		$c->add(FileSyncPeer::PARTNER_ID, $asset->getPartnerId());
+		$c->add(FileSyncPeer::STATUS, FileSync::FILE_SYNC_STATUS_READY);
+		$c->add(FileSyncPeer::FILE_TYPE, FileSync::FILE_SYNC_FILE_TYPE_URL);
+		
+		$sharedStorageProfileIds = kDataCenterMgr::getSharedStorageProfileIds();
+		if(count($sharedStorageProfileIds))
+		{
+			$c->add(FileSyncPeer::DC, $sharedStorageProfileIds, $sharedStorageProfileIds);
```

Comment:
@yossipapi  didn't you mean $c->add(FileSyncPeer::DC, $sharedStorageProfileIds, Criteria::IN); or something like that?

Reply:
Already fixed.